### PR TITLE
link pages

### PR DESCRIPTION
**Pull request by sepehr9877**
i guess now all of our pages are linked .we have three types of dashboard User ,Company and Admin .For Accessing  pages, you need to switch between dashboards . for example in order to see job posts  you have to go to User dashboard there is a link there then you can see all job posts

### CODE REVIEW

**Comment by MajorTreble** [comment]
75% coverage on new code, you are only missing 5% to get a green check mark

File: src/main/java/ca/concordia/soen6011/configuration/WebSecurityConfig.java [hunk 1]
```diff
@@ -18,6 +18,9 @@ public SecurityFilterChain securityFilterChain(HttpSecurity http) throws Excepti
         http
             .authorizeHttpRequests(requests -> requests
             		.requestMatchers("/", "/error", "/register").permitAll()
+
+                    .requestMatchers("/checkUsername","/company/submitinterview").permitAll()
```
Comment (MajorTreble) [inline src/main/java/ca/concordia/soen6011/configuration/WebSecurityConfig.java:22]: submit interview should not be unauthorized

File: src/main/java/ca/concordia/soen6011/controller/AdminController.java [hunk 2]
```diff
@@ -154,9 +155,9 @@ public String search_jobs(@RequestParam("query") String item,Model model) {
 	}
 
 	@GetMapping("/users")
-	public String manageUsers(Model model) {
-		List<User> users = userRepository.findAll();
-		System.out.println(users);  // Print users for debugging
+	public String manageUsers(Model model,Authentication authentication) {
```
Comment (MajorTreble) [inline src/main/java/ca/concordia/soen6011/controller/AdminController.java:158]: You can use UserDetails directly.

File: src/main/java/ca/concordia/soen6011/controller/UserContextControllerAdvice.java [hunk 3]
```diff
@@ -17,5 +22,16 @@ public String getCurrentUser(Authentication authentication) {
 		return username;
         
     }
+	@ModelAttribute("ROLE_ADMIN")
```
Comment (MajorTreble) [inline src/main/java/ca/concordia/soen6011/controller/UserContextControllerAdvice.java:25]: nice

File: src/main/java/ca/concordia/soen6011/repository/UserRepository.java [hunk 4]
```diff
@@ -18,6 +19,8 @@ public interface UserRepository extends JpaRepository<User,Integer> {
      * @param password password received from the view.
      * @return User object
      */
+	@Query("SELECT u FROM User u WHERE u.username <> :username")
+    List<User> findAllExceptUsername(String username);
```
Comment (MajorTreble) [inline src/main/java/ca/concordia/soen6011/repository/UserRepository.java:23]: You could try findByUserUsernameNot

**Review by nioushamahjoub — APPROVED** [review]

**Pull request merged**